### PR TITLE
Allow PKCS8 encoded EC private keys

### LINKED ISSUE #3283: OPA does not support EC PKCS8 keys
## Expected Behavior
OPA should be able to use EC PKCS8 private keys.

## Actual Behavior
OPA parses EC PKCS1 but fails parsing EC PKCS8 keys.

## Steps to Reproduce the Problem
Configure OAuth2 JWT auth with an EC PKCS8 PEM private key.

## Additional Info
Support for RSA PKCS8 keys were added in https://github.com/open-policy-agent/opa/pull/3117, but when configuring with an EC PKCS8 key the following error is returned:
`request failed: x509: failed to parse private key (use ParsePKCS8PrivateKey instead for this key format)`

Relevant code snippet:
https://github.com/open-policy-agent/opa/blob/99c81cab7c754f9b0f62e77a300f7455d3bcea2b/internal/jwx/jws/sign/sign.go#L47-L57

### PR DESCRIPTION
**Pull request by andrehaland**
Fixes #3283

Signed-off-by: Andre Håland <andre.haland@gmail.com>

### CODE REVIEW

File: bundle/keys_test.go [hunk 1]
```diff
@@ -191,11 +192,25 @@ HdZaQEvVwuOfmykc6fRJg3TTW2FyTZkr89Pt7gkffwKBgHGeJkFc6LFeHIwa3SbS
 qAVebnCAfNo9hHxz3xYA0PaCF3Kr1X9z4X2tF2Za7nWfVbfWViAncLrJgjnHRdrs
 f10hbJEuLFhD1c2dNjwqflANV5OanG1syqYqil5TgWm1AaRFj+PbRPk0FRfF9y+e
 tKaHBn4eyNlKjQaEn16ZxKJm
+-----END PRIVATE KEY-----`
+
+	pkcs1ecPrivateKey := `-----BEGIN EC PRIVATE KEY-----
```
Comment (anderseknert) [inline bundle/keys_test.go:197]: Wow, these keys are compact! 😄

**Review by anderseknert — APPROVED** [review]
We should probably parameterize these tests at some point, but this looks good to me!

File: internal/jwx/jws/sign/sign.go [hunk 2]
```diff
@@ -52,7 +52,11 @@ func GetSigningKey(key string, alg jwa.SignatureAlgorithm) (interface{}, error)
 
 		priv, err := x509.ParseECPrivateKey(block.Bytes)
 		if err != nil {
-			return nil, err
+			pkcs8priv, err2 := x509.ParsePKCS8PrivateKey(block.Bytes)
+			if err2 != nil {
+				return nil, fmt.Errorf("error parsing private key (%v), (%v)", err, err2)
+			}
+			return pkcs8priv, nil
```
Comment (srenatus) [inline internal/jwx/jws/sign/sign.go:59]: oh my dear, this function.... but it's OK, it's not on you. 😄

**Review by srenatus — APPROVED** [review]

**Comment by srenatus** [comment]
💭 Any docs needs?

**Comment by andrehaland** [comment]
> 💭 Any docs needs?

I guess the relevant place would be here https://www.openpolicyagent.org/docs/latest/configuration/#keys? Should I add a line about support for PKCS8 and PKCS1, or is it not needed?

**Comment by srenatus** [comment]
@andrehaland Yeah I had realised we hadn't done this for PKCS1 either. So, fine with this PR, thanks again.Happy to merge docs improvements, still! 😄

**Pull request merged**